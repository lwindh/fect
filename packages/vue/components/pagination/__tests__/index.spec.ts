import { mount } from '@vue/test-utils'
import { Pagination } from '..'

describe('Pagination', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Pagination)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be work error when in edge case', async () => {
    const wrapper = mount(Pagination, {
      props: {
        limit: 2,
        count: 10
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setProps({ count: 0 })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be support simple mode render', () => {
    const wrapper = mount(Pagination, {
      props: {
        simple: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be render all pages wehen limit value is older than the count value', () => {
    const wrapper = mount(Pagination, {
      props: {
        limit: 11,
        count: 10
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('button').length).toBeGreaterThanOrEqual(12)
  })
  it('should trigger change page number', async () => {
    const wrapper = mount(Pagination, {
      props: {
        modelValue: 1,
        count: 10
      }
    })
    const els = wrapper.findAll('.pagination-item__button ')
    expect(wrapper.vm.modelValue).toEqual(1)
    await els[2].trigger('click')
    expect(wrapper.emitted().change[0]).toEqual([2])
    await els[els.length - 1].trigger('click')
    expect(wrapper.emitted().change[1]).toEqual([3])
    await els[0].trigger('click')
    expect(wrapper.emitted().change[0]).toEqual([2])
  })
  it('should render as custom prev or next', async () => {
    const wrapper = mount({
      components: {
        [Pagination.name]: Pagination
      },
      data: () => ({ value: 1, count: 10, limit: 11, size: 'medium' }),
      template: `<fe-pagination v-model="value" :count="count" 
      :limit="limit" :size="size">
       <template #prev>
        上一页
       </template>
       <template #next>
        下一页
       </template>
      </fe-pagination>`
    })

    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setData({ limit: 7 })
    const els = wrapper.findAll('.paginatuon-slots__custom')
    expect(wrapper.html()).toMatchSnapshot()
    await els[els.length - 1].trigger('click')
    expect(wrapper.vm.value).toEqual(2)
    // eslint-disable-next-line prefer-destructuring
    const ellipsisEl = wrapper.findAll('.pagination-item__button ')[7]
    await ellipsisEl.trigger('mouseenter')
    await ellipsisEl.trigger('click')
    await ellipsisEl.trigger('mouseleave')
    expect(wrapper.vm.value).toEqual(7)
  })
})
