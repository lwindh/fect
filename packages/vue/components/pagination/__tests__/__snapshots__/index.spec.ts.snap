// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should be render all pages wehen limit value is older than the count value 1`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Prev</button></li>
    <li><button class="pagination-item__button active " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">1</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">2</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">3</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">4</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">5</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">6</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">7</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">8</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">9</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">10</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Next</button></li>
  </ul>
</nav>
`;

exports[`Pagination should be render as a element 1`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Prev</button></li>
    <li><button class="pagination-item__button active " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">1</button></li>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Next</button></li>
  </ul>
</nav>
`;

exports[`Pagination should be support simple mode render 1`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li><button class="pagination-simple__side disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Prev</button></li>
    <li class="pagination-simple__desc">1 / 1</li>
    <li><button class="pagination-simple__side disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Next</button></li>
  </ul>
</nav>
`;

exports[`Pagination should be work error when in edge case 1`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Prev</button></li>
    <li><button class="pagination-item__button active " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">1</button></li>
    <!---->
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">2</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);"><svg viewBox="0 0 24 24" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
          <circle cx="12" cy="12" r="1" fill="currentColor"></circle>
          <circle cx="19" cy="12" r="1" fill="currentColor"></circle>
          <circle cx="5" cy="12" r="1" fill="currentColor"></circle>
        </svg></button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">10</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Next</button></li>
  </ul>
</nav>
`;

exports[`Pagination should be work error when in edge case 2`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Prev</button></li>
    <li><button class="pagination-item__button disabled " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">Next</button></li>
  </ul>
</nav>
`;

exports[`Pagination should render as custom prev or next 1`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li class="paginatuon-slots__custom"> 上一页 </li>
    <li><button class="pagination-item__button active " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">1</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">2</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">3</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">4</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">5</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">6</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">7</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">8</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">9</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">10</button></li>
    <li class="paginatuon-slots__custom"> 下一页 </li>
  </ul>
</nav>
`;

exports[`Pagination should render as custom prev or next 2`] = `
<nav class="fect-pagination" style="font-size: 14px; --pagination-size: 32px;">
  <ul>
    <li class="paginatuon-slots__custom"> 上一页 </li>
    <li><button class="pagination-item__button active " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">1</button></li>
    <!---->
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">2</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">3</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">4</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">5</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">6</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">7</button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);"><svg viewBox="0 0 24 24" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
          <circle cx="12" cy="12" r="1" fill="currentColor"></circle>
          <circle cx="19" cy="12" r="1" fill="currentColor"></circle>
          <circle cx="5" cy="12" r="1" fill="currentColor"></circle>
        </svg></button></li>
    <li><button class="pagination-item__button  " style="--pagination-hover: rgba(0, 0, 0, 0.1); --pagination-activeHover: rgba(0, 0, 0, 0.8);">10</button></li>
    <li class="paginatuon-slots__custom"> 下一页 </li>
  </ul>
</nav>
`;
